api/swagger.yaml

```diff
@@ -0,0 +1,282 @@
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/error'
+components:
+  parameters:
+    user_id:
+      name: user_id
+      in: query
+      description: User ID
+      required: true
+      schema:
+        type: string
+    chat_id:
+      name: chat_id
+      in: path
+      description: Chat ID
+      required: true
+      schema:
+        type: string
+    last_message_id:
+      name: last_message_id
+      in: query
+      description: Id of the last message already received. If set, the endpoint only returns newer messages
+      schema:
+        type: string
+  schemas:
+    chat_list_response:
+      type: object
+      properties:
+        chats:
+          type: array
+          items:
+            $ref: '#/components/schemas/detailed_chat'
+    chat_creation:
+      type: object
+      properties:
+        user_ids:
+          type: array
+          items:
+            type: string
+          example: [ 123, 321 ]
+        type:
+          $ref: '#/components/schemas/chat_type'
+    chat_response:
+      $ref: '#/components/schemas/detailed_chat'
+    detailed_chat:
+      allOf:
+        - $ref: '#/components/schemas/base_chat'
+        - $ref: '#/components/schemas/message_list_response'
+        - type: object
+          properties:
+            last_message:
+              $ref: '#/components/schemas/message'
+    base_chat:
+      properties:
+        chat_id:
+          type: string
+          example: 543
+        user_ids:
+          type: array
+          description: Ids of the users participating in the chat
+          items:
+            type: string
+          example: [ 123, 321 ]
+        last_activity:
+          type: string
+          format: date-time
+          description: Datetime of the last activity from this chat
+        type:
+          $ref: '#/components/schemas/chat_type'
+    chat_type:
+      type: string
+      enum:
+        - group
+        - direct
+    new_message_list_response:
+      type: object
+      properties:
+        new_messages:
+          type: array
+          items:
+            allOf:
+              - type: object
+                properties:
+                  chat_id:
+                    type: string
+                    example: 123
+              - $ref: '#/components/schemas/message_list_response'
+    message_list_response:
+      type: object
+      properties:
+        messages:
+          type: array
+          items:
+            $ref: '#/components/schemas/message'
+    message:
+      allOf:
+        - $ref: '#/components/schemas/base_message'
+        - type: object
+          properties:
+            id:
+              type: string
+              description: id of the message
+              example: 1234
+            is_read:
+              type: boolean
+              example: true
+              description: |
+                Defines if the recipient has already read the message
```

Comment:
Same as with new:
this object ist being used in a group chat, with a relation to a particular user.
So what does it mean?
That it is new to all group members (even if one out of 10 is missing).
This is not what you intend to do, I guess.
You want to put it in relation to a certain user.
What about:
let the chat-service return a map (User->is_readflag).
The BBF will sort this out when returning the content to the angular app?

Reply:
The resource is requested here for a user. So both properties are meant only for this specific user as they do not make sense based on a group. In the domain model they are therefore also not part of the message resource but the participants resource. So the participant resource has the mapping chat_id, user_id and data about the last read, last received and last notified message. For the UI is does not matter what other users additionally haven't read a message but only if the user who requests the message haven't read it.

Reply:
Then I a missing anything in the request, that has a reference to the user.
If the chat dancer is running behind a BBF, it will not get this from the token, right (even if we would use authentication here it all, would we use service-token don't we)?
Or would you propose to forward the token from the browser?
Let's see today when we meet.

Reply:
With regards to the api of the dancer:
hiding the user-information for eg. the profile endpoint in the token, made the endpoint less useful. I realized it just this week, when I wrote an importer for the test-date, xiaofei is working on:
whenever I wanted to update a profile from a user, I am forced to login as this user and can not use eg. an admin-account to perform such requests.
I am not sure if we want to continue with this kind of API  ;-)
See you afternoon!

Reply:
In regards to the question how to identify the user. I use the query parameter user_id (see in the endpoints) ;)